app/controllers/users_controller.rb

```diff
@@ -280,27 +266,19 @@ def visible_items(current_user)
     #       .visible_to_registered_user.
     visible_method = current_user.nil? && current_admin.nil? ? :visible_to_all : :visible_to_registered_user
 
-    # hahaha omg so ugly BUT IT WORKS :P
+    visible_works = @user.works.send(visible_method)
+    visible_series = @user.series.send(visible_method)
+    visible_bookmarks = @user.bookmarks.send(visible_method)
+
+    visible_works = visible_works.revealed.non_anon
     @fandoms = if @user == User.orphan_account
                  []
                else
-                 Fandom.select('tags.*, count(tags.id) as work_count').
-                   joins(:direct_filter_taggings).
-                   joins("INNER JOIN works
-                     ON filter_taggings.filterable_id = works.id
-                     AND filter_taggings.filterable_type = 'Work'").
-                   group('tags.id').
-                   merge(Work.send(visible_method).revealed.non_anon).
-                   merge(Work.joins("INNER JOIN creatorships
-                     ON creatorships.creation_id = works.id
-                     AND creatorships.creation_type = 'Work'
-                     INNER JOIN pseuds ON creatorships.pseud_id = pseuds.id
-                     INNER JOIN users ON pseuds.user_id = users.id").
-                   where('users.id = ?', @user.id))
+                 Fandom.select("tags.*, count(DISTINCT works.id) as work_count").
```

Comment:
This creates invalid SQL for me. As soon as we use `joins(:filtered_works)` or `.merge(visible_works)`, Rails adds `DISTINCT works.*` to the select.

Reply:
Are you running it on master, or in the creatorships branch? I cleaned up the definition of the user `works` association as part of this PR.

Reply:
Ha, good catch. I ran it on master, without any of the other fixes.